contrib/win32/win32compat/termio.c

```diff
@@ -260,6 +260,18 @@ int
 syncio_close(struct w32_io* pio)
 {
 	debug4("syncio_close - pio:%p", pio);
+
+	/* Flush descriptor.*/
```

Comment:
Similar justification as: https://github.com/PowerShell/openssh-portable/blob/cdee73645a2c4506fa8c73835f55d14ed196a980/contrib/win32/win32compat/fileio.c#L1089-L1100

```suggestion
	/*
        * Wait for io write operation that is called by worker thread to terminate
        * to avoid the write operation being terminated prematurely by CancelIoEx.
	* If you see any process waiting here indefinitely - its because no one
	* is draining from other end of the pipe. This is an unfortunate
	* consequence that should otherwise have very little impact on practical
	* scenarios.
	*/
```

Feel free to modify this comment with any details that you think is relevant.

Reply:
This comment is similar to alternate path code for the file close operation, so I think it makes sense have it here as well, since we are using the same wait pattern.

Reply:
Yes, this was basically to perform the same logic before and after. My understanding/recollection is the sleep is actually needed to ensure that the queue actually gets drained.

Reply:
I'm just wondering if the queue needs to be drained immediately after the write operation terminates, or if we can just have one SleepEx call once both the read and write operation have terminated. I don't think it hurts to have it there, so I am good to merge as is. But I'm just wondering if it's needed.